Quadriga/src/main/java/edu/asu/spring/quadriga/web/workbench/DeleteProjectController.java

```diff
@@ -1,164 +1,54 @@
 import edu.asu.spring.quadriga.aspects.annotations.ElementAccessPolicy;
-import edu.asu.spring.quadriga.domain.factory.workbench.IModifyProjectFormFactory;
 import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
 import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
 import edu.asu.spring.quadriga.service.workbench.IModifyProjectManager;
-import edu.asu.spring.quadriga.validator.ProjectFormValidator;
 import edu.asu.spring.quadriga.web.login.RoleNames;
-import edu.asu.spring.quadriga.web.workbench.backing.ModifyProject;
-import edu.asu.spring.quadriga.web.workbench.backing.ModifyProjectForm;
-import edu.asu.spring.quadriga.web.workbench.backing.ModifyProjectFormManager;
 
 @Controller
-public class DeleteProjectController 
-{
-	
-	@Autowired
-	private IModifyProjectFormFactory projectFormFactory; 
-	
+public class DeleteProjectController {
 	@Autowired
 	IModifyProjectManager projectManager;
-	
-	@Autowired
-	IProjectSecurityChecker projectSecurity;
-	
-	@Autowired
-	ProjectFormValidator validator;
-	
-	@Autowired
-	ModifyProjectFormManager projectFormManager;
-	
-	private static final Logger logger = LoggerFactory
-			.getLogger(DeleteProjectController.class);
-	
-	@InitBinder
-	protected void initBinder(WebDataBinder validateBinder) throws Exception 
-	{
-		
-		validateBinder.setValidator(validator);
-	}
-	
+
 	/**
-	 * This method is called during form load to display all projects.
-	 * @param   model
-	 * @param   principal
-	 * @return  String - URL on success or failure.
-	 * @throws  QuadrigaStorageException
-	 * @author  Kiran Kumar Batna
+	 * This method calls the user manager to delete the project.
+	 * 
+	 * @param projectId Project identifier corresponding to project to be deleted
+	 * @return model - URL on success and failure.
+	 * @throws QuadrigaStorageException
+	 * @throws QuadrigaAccessException
 	 */
-	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT,paramIndex = 0, userRole = {RoleNames.ROLE_COLLABORATOR_ADMIN} )})
-	@RequestMapping(value="auth/workbench/deleteproject", method=RequestMethod.GET)
-	public ModelAndView deleteProjectRequestForm(Principal principal) throws QuadrigaStorageException,QuadrigaAccessException
-	{
-		ModelAndView model;
-		String userName;
-		ModifyProjectForm projectForm;
-		List<ModifyProject> deleteProjectList;
-		
-		//create a model
-		model = new ModelAndView("auth/workbench/deleteproject");
-		userName = principal.getName();
-		
-		//create a project form
-		projectForm = projectFormFactory.createModifyProjectFormObject();
-		
-		deleteProjectList = projectFormManager.getProjectList(userName,RoleNames.ROLE_COLLABORATOR_ADMIN);
-		
-		projectForm.setProjectList(deleteProjectList);
-		
-		//adding the model
-		model.getModelMap().put("projectform", projectForm);
-		model.getModelMap().put("success", 0);
+	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 0, userRole = { RoleNames.ROLE_COLLABORATOR_ADMIN }) })

```

Comment:
The paramIndex should not be 0; userRoles should also include project admin (owner of project).